public/app/features/explore/state/actions.ts

```diff
@@ -530,6 +533,124 @@ export const runQueries = (exploreId: ExploreId): ThunkResult<void> => {
   };
 };
 
+/**
+ * Sub-action to run append queries
+ */
+export const runAppendQueries = (exploreId: ExploreId, direction: QueryDirection): ThunkResult<void> => {
+  return (dispatch, getState) => {
+    dispatch(updateTime({ exploreId }));
+
+    const exploreItemState = getState().explore[exploreId];
+    const {
+      datasourceInstance,
+      queries,
+      containerWidth,
+      isLive: live,
+      range,
+      scanning,
+      queryResponse,
+      querySubscription,
+      mode,
+      showingGraph,
+      showingTable,
+    } = exploreItemState;
+
+    const newestResponseEndTimestamp = queryResponse.series[0].fields[0].values.get(
+      Math.floor((queryResponse.series[0].fields[0].values.length - 1) / 2)
+    );
+
+    const oldestResponseEndTimestamp = queryResponse.series[queryResponse.series.length - 1].fields[0].values.get(
+      Math.floor((queryResponse.series[0].fields[0].values.length - 1) / 2)
+    );
```

Comment:
Not sure I understand these 2. Why are we taking the value from the middle? If I understand it that will mean we will get half of the rows again from the new response don't we?
`queryResponse.series[queryResponse.series.length - 1]` what are the additional series. Feels like it would make sense if those are just older logs, but not sure if Loki for example does not return each stream as different series as we need to sort it later on.